lib/effects/effects.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/animation.dart';
+import 'package:meta/meta.dart';
+
+import 'dart:math';
+import 'dart:ui';
+
+import '../position.dart';
+import '../components/component.dart';
+
+abstract class PositionComponentEffect {
+  void update(double dt);
+  bool hasFinished();
+  PositionComponent component;
+}
+
+double _direction(double p, double d) => p < d ? -1 : 1;
```

Comment:
this is equivalent to (p - d).sign(), which I think is nice because it's very parallel to the impl of _size